### ISSUE TITLE
Kinesis enhanced fan-out consumers

### ISSUE THREAD

**Issue by gee-m** [redpanda-data/connect#1401]
Hey there, love the tool!

Just a feature suggestion in case more people are interested than just me - it would be cool to support [kinesis enhanced fan-out consumers](https://docs.aws.amazon.com/streams/latest/dev/building-enhanced-consumers-api.html). As far as the research I've done, here's some notes hopefully that can help see clearer into this task

- It would require some change as we use [`GetShardIteratorWithContext`](https://github.com/benthosdev/benthos/blob/7c6a9a5eb39724ca4a7ff318f15bb6375eb338c5/internal/impl/aws/input_kinesis.go#L209) but would need to use [`SubscribeToShard`](https://docs.aws.amazon.com/sdk-for-go/api/service/kinesis/#Kinesis.SubscribeToShard) with `SubscribeToShardEvent`s
- It could be a new input?
- Could use the [VMWare kinesis library ](https://github.com/vmware/vmware-go-kcl/pull/90)to make it easier
- Ability to either `RegisterNewStreamConsumer` or use an existing one (e.g. defined in terraform) would be cool

Here's some resources:
- [AWS Go docs](https://docs.aws.amazon.com/sdk-for-go/api/service/kinesis/#Kinesis.RegisterStreamConsumer)
- [AWS Usage example (java)](https://docs.aws.amazon.com/streams/latest/dev/building-enhanced-consumers-api.html)

**Comment by saholman** [comment]
This is something I'm going to need in the near future. @Jeffail are there plans to add this to the existing Kinesis input? If I implement a new consumer that supports Kinesis enhanced fan-out, is that something you'd be interested in merging back into the baseline?

**Comment by Jeffail** [comment]
@saholman yeah I'd be happy to take a PR, ideally something interacting with the AWS client libraries directly rather than via another dependency, but happy to take a look at anything.

**Comment by michael-pressburg-cko** [comment]
Are there any updates on this? @Jeffail

**Comment by twmb** [comment]
Closing as stale — Jeffail indicated willingness to accept a PR for enhanced fan-out support, but no implementation materialized. If someone wants to contribute this, feel free to open a new issue or PR.
